### PR TITLE
Location Field: Support mode icons

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**

Mode icons were being ignored, all stops were showing up as a bus. This is resolved by implementing a prop!

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/components/form/connected-location-field.tsx [hunk 1]
```diff
@@ -23,6 +26,31 @@ const ConnectedLocationField = connectLocationField(StyledLocationField, {
...
+  // @ts-expect-error No type on ComponentContext
+  const { ModeIcon } = useContext(ComponentContext)
+
+  const { properties } = feature
+  if (feature && properties) {
+    const { modes } = properties
+    if (modes && modes.length > 0) {
+      return (
+        <IconWrapper>
+          <ModeIcon mode={modes[0].toLowerCase()} role="img" />
```
Comment (amy-corson-ibigroup) [inline lib/components/form/connected-location-field.tsx:46]: If you've got a `role="img"` on here, I think you'll need an `aria-label` too! If they're decorative, you can probably just `aria-hidden`

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Just one small a11y thing

File: lib/components/form/connected-location-field.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/form/connected-location-field.tsx:46]: Amazing catch thank you!

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Could you address the ambiguous behavior about mode icons?

File: lib/components/form/connected-location-field.tsx [hunk 2]
```diff
@@ -23,6 +26,31 @@ const ConnectedLocationField = connectLocationField(StyledLocationField, {
...
+  // @ts-expect-error No type on ComponentContext
+  const { ModeIcon } = useContext(ComponentContext)
+
+  const { properties } = feature
+  if (feature && properties) {
+    const { modes } = properties
+    if (modes && modes.length > 0) {
+      return (
+        <IconWrapper>
+          <ModeIcon aria-hidden mode={modes[0].toLowerCase()} role="img" />
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/form/connected-location-field.tsx:46]: No! I think it's clear by what's in the brackets that this is a stop

[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/form/connected-location-field.tsx:46]: @miles-grant-ibigroup Then can you remove the `role="img"` please? From what you replied above, the mode icon is purely decorative and can stay hidden from screen readers.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
See https://github.com/opentripplanner/otp-react-redux/pull/1130/files#r1484695345

File: lib/components/form/connected-location-field.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/form/connected-location-field.tsx:46]: Is there any chance that the role is affecting presentation of the svg? When I remove the prop the icons no longer appear correctly (Safari)

**Comment by miles-grant-ibigroup** [comment]
Comment added

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**